cmd/compute-domain-daemon/computedomain.go

```diff
@@ -243,6 +250,38 @@ func (m *ComputeDomainManager) UpdateComputeDomainNodeInfo(ctx context.Context,
 	return nil
 }
 
+// getNextAvailableIndex finds the next available index for a new node,
+// filling gaps before extending the sequence. Returns an error if no
+// index is available within maxNodesPerIMEXDomain.
+func getNextAvailableIndex(nodes []*nvapi.ComputeDomainNode, maxNodesPerIMEXDomain int) (int, error) {
+	if len(nodes) >= maxNodesPerIMEXDomain {
+		return -1, fmt.Errorf("cannot add more nodes, already at maximum (%d)", maxNodesPerIMEXDomain)
+	}
+
+	// Create a map to track used indices
+	usedIndices := make(map[int]bool)
+
+	// Collect all currently used indices
+	for _, node := range nodes {
+		if node.Index >= 0 && node.Index < maxNodesPerIMEXDomain {
```

Comment:
Reading up to here, I think I'd appreciate a comment that explains why `node.Index` can be outside of the `[0, max...]` interval (i.e., what that means)?

Reply:
If constructed properly it can't. I could drop this check if it seems redundant or otherwise wrap it in a runtime.Must() so that it panics if it's not true.

Reply:
> wrap it in a runtime.Must() so that it panics if it's not true.

Perfect.

I like doing these types of "assertions" a lot, probably also inspired because of writing in math/physics code.

That expresses the intent (helps readability), maintains debuggability.